executor/analyze.go

```diff
@@ -61,10 +61,14 @@ func (e *AnalyzeExec) Next(ctx context.Context, req *chunk.RecordBatch) error {
 		return err
 	}
 	taskCh := make(chan *analyzeTask, len(e.tasks))
-	resultCh := make(chan statistics.AnalyzeResult, len(e.tasks))
+	resultCh := make(chan analyzeResult, len(e.tasks))
 	for i := 0; i < concurrency; i++ {
 		go e.analyzeWorker(taskCh, resultCh)
 	}
+	statistics.ClearHistoryJobs()
```

Comment:
How about to maintain the latest N finished/failed analyze jobs, auto remove the the stale history jobs?

For now, we print all the jobs for `show analyze status`. It would be strange from the user side that the history info disappeared when a `analyze table` is executed.

| timeline | user 1 | user 2 |
|--|--|--|
| 1 | analyze table **t1**; | |
| 2 | show analyze status; | |
| 3 | have history results about **t1** | |
| 4 | | analyze table **t2**;|
| 5 | show analyze status; | |
| 6 | no history results about **t1**| |